### PR TITLE
Improves CTLOG_STORE setters

### PR DESCRIPTION
**Pull request by RJPercival**
Changes them to have clearer ownership semantics, as suggested in
https://github.com/openssl/openssl/pull/1372#discussion_r73232196.

### CODE REVIEW

**Comment by richsalz** [comment]
+1

File: crypto/ct/ct_policy.c [hunk 1]
```diff
@@ -30,21 +30,25 @@ CT_POLICY_EVAL_CTX *CT_POLICY_EVAL_CTX_new(void)
...
+    X509_free(ctx->cert);
+    X509_free(ctx->issuer);
     OPENSSL_free(ctx);
 }
 
-void CT_POLICY_EVAL_CTX_set0_cert(CT_POLICY_EVAL_CTX *ctx, X509 *cert)
+void CT_POLICY_EVAL_CTX_set1_cert(CT_POLICY_EVAL_CTX *ctx, X509 *cert)
 {
-    ctx->cert = cert;
+    if (X509_up_ref(cert))
```
Comment (ekasper) [inline crypto/ct/ct_policy.c]: If they can fail then they should no longer be void.

Comment (RJPercival) [inline crypto/ct/ct_policy.c]: Done.

**Comment by richsalz** [comment]
still +1 needs another reviewer.

File: util/libcrypto.num [hunk 2]
```diff
@@ -291,7 +291,7 @@ CRYPTO_gcm128_setiv                     291	1_1_0	EXIST::FUNCTION:
 ASN1_PCTX_set_oid_flags                 292	1_1_0	EXIST::FUNCTION:
 d2i_ASN1_INTEGER                        293	1_1_0	EXIST::FUNCTION:
 i2d_PKCS7_ENCRYPT                       294	1_1_0	EXIST::FUNCTION:
-CT_POLICY_EVAL_CTX_set0_issuer          295	1_1_0	EXIST::FUNCTION:CT
+CT_POLICY_EVAL_CTX_set0_issuer          295	1_1_0	NOEXIST::FUNCTION:
```
Comment (ekasper) [inline util/libcrypto.num]: you could delete these since they now never existed (not sure if we have a script to clean them up before the release)

**Comment by ekasper** [comment]
+1

File: util/libcrypto.num (hunk 2 above)
Comment (RJPercival) [inline util/libcrypto.num]: Should I leave a gap in the number sequence or replace them with the new set1 functions?

Comment (richsalz) [inline util/libcrypto.num]: Put the new functions directly in place of the old ones.  Then I’ll merge.

Comment (RJPercival) [inline util/libcrypto.num]: Done.

**Comment by richsalz** [comment]
merged, thanks.  closing.

**Pull request closed without merging**